lib/processor.rb

```diff
@@ -33,5 +38,30 @@ def event_model(type)
     def event_type(event)
       event.fetch("event_type", nil)
     end
+
+    def geolocate(event)
+      return unless event["_ip_address"]
+
+      result = Geocoder.search(event["_ip_address"])
+
+      if match = result[0]
+        event.merge(geo: {
+          country_name: match.country,
+          country_code: match.country_code,
+          city_name: match.city,
+          coordinates: [match.longitude, match.latitude]
```

Comment:
This `coordinates` array is how ElasticSearch needs it. Happy to remove the separate lat/long, I decided to leave it in to make the JSON self-documenting (otherwise you can't easily know if coordinates is `[long, lat]` or `[lat, long]`).